### DISCUSSION TITLE
How to use SSG with multilingual?

### DISCUSSION THREAD

**Discussion by manhthepixta** [vercel/next.js#16262] [Help]
Currently I'm using SSG which is perfect. But I want my site support multilingual. I tried next-18next but it doesn't support SSG now.
I'm thinking using local subpath (e.g `domain.com` and `domain.com/es`) but I cannot imagine how to generate a new set of static html pages of the new language.
Does anyone have any idea?

**Comment by tjperry07** [comment]
https://github.com/vercel/next.js/discussions/16267 Linking the conversations because I might be able to use the answer here for my issue.

**Comment by andrewfairlie** [comment]
One way we're exploring is having a URL structure like...

```
[language]/index.js
[language]/blog/[slug]
```

With this method we do the rest of the templates as if they're English but within the `[language]` folder. You can then do two things:-

1. Fallback to an error if it fails to fetch that entry from your CMS in that language (this works for our approach with Craft CMS but your results may vary)
2. Rewrite and redirect URLs if the paths need to be different in different languages. For example you can rewrite `/de/contact` to `/de/kontact` and then also add a redirect to `/de/contact/` to `/de/kontact` to avoid duplications.
